lib/src/device.dart

```diff
@@ -210,4 +211,48 @@ class UdevDevice {
       mapHash(sysattrs),
     );
   }
+
+  R _usingPointer<R>(R Function(ffi.Pointer<udev_device>) computation) {
+    final ptr = ffi.using((arena) {
+      final csyspath = syspath.toCString(allocator: arena);
+      final ctx = UdevContext();
+      final ptr = dylib.udev_device_new_from_syspath(ctx.toPointer(), csyspath);
```

Comment:
I wanted to make UdevDevice a plain value class to avoid keeping a native udev_device instance around but this makes me wonder if we should reconsider that make the UdevDevice class a wrapper around the native udev_device instance and rely on finalizers for clean-up. If we end up re-creating it anyway, might as well just keep it... :smile:

Reply:
Good to know